File: leetcode/python/1496_path_crossing.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        cur = [0, 0]
        visited = set()
        for c in path:
            visited.add(tuple(cur))
            if c == 'N':
                cur[1] += 1
            elif c == 'S':
                cur[1] -= 1
            elif c == 'E':
                cur[0] += 1
            else:
                cur[0] -= 1
            if tuple(cur) in visited:
                return True
        return False
        

File: leetcode/rust/src/1496_path_crossing.rs
use std::collections::HashSet;

impl Solution {
    pub fn is_path_crossing(path: String) -> bool {
        let mut pos = (0, 0);
        let mut visited = HashSet::new();
        for c in path.chars() {
            visited.insert(pos);
            match c {
                'N' => pos.1 += 1,
                'S' => pos.1 -= 1,
                'E' => pos.0 += 1,
                'W' => pos.0 -= 1,
                _ => (),
            }
            if visited.contains(&pos) {
                return true;
            }
        }
        
        false
    }
}